### DISCUSSION TITLE
Implementation and Doubt

### DISCUSSION THREAD

**Discussion by ghost** [SethRobinson/UGT#46] [Ideas]
1st Good afternoon, I would like to know if it is possible to use the https://ztranslate.net/ API for the ORC and translation functions, I believe that one project ends up complementing the other. both projects are admirable.

2nd When you use the google API, has your credit card been charged for using google?

Thanks!

**Comment by SethRobinson** [comment]
It looks like to me ztranslate.net also uses google apis under the hood ("automatic mode"), unless "packages" are used, a kind of special cached version where people can adjust/write custom translations - however, it's not clear to me if packages can be used easily by third party clients, probably not?

I've personally never had to pay for using the Google API as I've not gone over the monthly limits, but it could happen.

My work on UGT is limited to what I personally need it for, but I'm happy to apply source changes from other people who want to add stuff.

**Comment by ghost** [comment]
Got it, could you tell me what the monthly limits are? I didn't understand the google billing method an API can be charged for different usage methods this is very confusing at first glance.

**Reply by SethRobinson** [reply]
I don't remember, but it's something like one million characters of translated text per month or something.  (the image to text and text to speech functions have limits I don't remember as well)

I setup an ["email me if I hit $5" alert to be safe](https://github.com/SethRobinson/UGT/issues/43#issuecomment-1248756016).
